python/paddle/fluid/dygraph/nn.py

```diff
@@ -1442,6 +1442,17 @@ def weight(self, value):
         self._w = value
 
     def forward(self, input):
+        if in_dygraph_mode():
```

Comment:
Some suggestions: move some common codes outside `if in_dygraph_modes()` for better maintainability in the future. Seems too many duplicate codes.

Reply:
OK, thanks a lot.